src/Storage.h

```diff
@@ -21,6 +21,7 @@ class Storage
   virtual byte readBytes(unsigned int eeaddress, byte nbytes, byte dest[]) = 0;
   virtual void writeBytes(unsigned int eeaddress, const byte src[], byte numbytes) = 0;
   virtual void reset() = 0;
+  virtual void commitWriteEEPROM() = 0;
```

Comment:
This line says that all child classes of Storage MUST override commitWriteEEPROM(). 
I am surprised you could compile as the other child classes are compiled even if they are not used in your project.

Reply:
Oh, sorry, I am wrong. I remember now how to do default implementation of virtual functions. We put the default implementation in the interface class (Storage) and then there is no need to change any child classes. 
So Storage.h should include:
`virtual void commitWriteEEPROM() {}`
Note the {} at the end. And no "override" here. Then you can revert the changes in the child classes.